scRNA-seq-advanced/setup/intro-setup.Rmd

```diff
@@ -0,0 +1,272 @@
+```{r}
+# Filter ADT
+# http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#applying-custom-qc-filters
+
+controls <- grep("^Ig", rownames(altExp(filtered_sce)))
+adt_qc_df <- DropletUtils::cleanTagCounts(altExp(filtered_sce), controls=controls) # I feel like this control might be crap!
+
+summary(adt_qc_df$high.controls) # these are all NA????????
+summary(adt_qc_df$discard) # these are MOSTLY NA????????
+
+# Let's do this - remove cells that have zero ambient concentrations:
+# This removes 11 cells
+cells_to_retain <- which(adt_qc_df$zero.ambient == FALSE)
+filtered_sce <- filtered_sce[, cells_to_retain]
+dim(filtered_sce)
+```
+
+
+
+### Normalize RNA and ADT:
+
+
+```{r}
+# Normalize RNA
+qclust <- scran::quickCluster(filtered_sce)
+filtered_sce <- scran::computeSumFactors(filtered_sce, clusters = qclust)
+norm_sce <- scater::logNormCounts(filtered_sce)
+
+# Normalize ADT
+
+# Calculate baseline:
+# Try out control-based normalization here 
+#  http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#control-based-normalization
+control_sf <- scater::librarySizeFactors(altExp(norm_sce), subset_row=controls)  # `controls` was previously defined during filtering
+sum(control_sf == 0) # _HOLY WOW_
+
+# Go for median instead, BUT TODO: would we want to exclude control here during baseline calculation? 
+baseline <- DropletUtils::ambientProfileBimodal(altExp(filtered_sce)) # !! currently this does not exclude controls
+median_sf <- scater::medianSizeFactors(altExp(norm_sce), reference = baseline)
+sum(median_sf == 0) # none are 0 here, soooo let's do this?
+
+# Normalize with the median size factors
+altExp(norm_sce) <- scater::logNormCounts(altExp(norm_sce), 
+                                          size.factors = median_sf)
+
+```
+
+### Behold, the filtered and normalized SCE:
+
+```{r}
+norm_sce
+
+altExp(norm_sce)
+```
+
+## Dimension reduction
+
+HVGs:
+```{r}
+num_genes <- 5000 # eh?
+
+gene_variance <- scran::modelGeneVar(norm_sce)
+hvg_list <- scran::getTopHVGs(gene_variance,
+                               n = num_genes)
+```
+
+PCA:
+```{r}
+# Do we want to set the number of PCs?
+norm_sce <- scater::runPCA(norm_sce,
+                           subset_row = hvg_list)
+```
+
+UMAP:
+```{r I map you map we all map for MapMap}
+# We can arbitrarily cluster first just to have some colors in there for funsies
+k <- 10
+
+# perform the clustering
+kclusters <- clusterRows(
+  reducedDim(norm_sce, "PCA"),
+  KmeansParam(centers = k)
+)
+
+# pop back into sce
+norm_sce$kcluster <- factor(kclusters)
+
+
+# Add in UMAPs and plot with colors
+norm_sce <- scater::runUMAP(norm_sce,
+                            dimred = "PCA")
+scater::plotReducedDim(norm_sce, "UMAP", colour_by = "kcluster")
+```
+
+## Cell type annotation
+
+We can introduce annotation with `SingleR`, and it might be a good idea to show how the results are highly influenced by the reference. 
```

Comment:
Before SingleR, we might want an example of annotation with ADT tags. At least we should plot an ADT tag onto the UMAP.

Reply:
This is something I haven't been sure about, for a couple reasons..
- Trying to find a good reproducible resource (as in, not via googling each protein name) for what celltypes the tags are roughly markers for
- Also a method has been elusive... or is the idea just to say "highest expressed tag wins" and call it a day?

Reply:
I would start with what Josh suggested and do something like plot CD3 and CD19 and say "look these should be B cells" and then with CD4 and CD8 "these should be T cells". I feel like that should be plenty here. 

From the [product sheet of the antibody panel](https://d1spbj2x7qk4bg.cloudfront.net/en-us/products/totalseq-b-human-tbnk-cocktail-19043?pdf=true&displayInline=true&leftRightMargin=15&topBottomMargin=15&filename=TotalSeq%E2%84%A2-B%20Human%20TBNK%20Cocktail.pdf&v=20220922120634) you should theoretically be able to see some T cells, B cells, NK cells and Monocytes.